add the RHICf library (for only storing raw data in BFC level)

ggfdsa10 opened the pull request:
1. new 2 makers 
a) StRHICfDbMaker (RHICf calibration data table maker)
b) StRHICfRawHitMaker (storing raw data maker)

2. modified StEvent, StBFChain and add some new RHICf class 
The revised classes are essential for makers to add later. (RHICf calibration and reconstruction makers)

3. new MuRHICf class (for store in MuDst) and some MuDst base class modifying

4. add the StRHICfUtil folder and class (there to be added another reconstruction tool)

Commit:
new makers for RHICf

Commit:
new RHICf data structrue

Commit:
add the part of RHICf

Commit:
add the new MuRHICf class

Commit:
modified class for RHICf muDst

Commit:
RHICf useful class (this folder to be add other RHICf class)

starsdong commented:
Hi Seunghwan,

Thank you for putting this PR up. Just to clarify how we proceed with the review.

1) These requested updates will go through the designated reviews
     BFChain:   Gene (@genevb)
     StEvent:  Jason (@klendathu2k)
     MuDstMaker:    Daniel B. (@jdbrice)

2) StRHICfDbMaker and StRHICfRawHitMaker are two new makers. We should go through the standard code review process. I will ask a couple of our colleagues to help on this (follow up on emails)

3) StRHICfUtil directory
     I am wondering whether you need this for production (meaning daq->MuDst) or they are used at the analyses level?

Thanks

ggfdsa10 commented:
Hello Xin,

Thank you for your follow-up.
Mostly, StRHICfUtil works at the analysis level (calibration and reconstruction).
But, There is some function (such as the tower size checker in StRHICfFunction in StRHICfUtil folder).
This tiny function (this name is checkGSOBarSize() ) is used in many parts such as StRHICfRawHitMaker, StMuRHICfUtil, etc.

so I would like to add this directory. 

Thank you.

P.S:
Also, StRHICfFunction includes the RHICf run type checker. 
this function is very important for reconstruction.

starsdong commented:
Thanks to Dmitry A. and Jeff for help review the new makers (StRHICfDbMaker and StRHICfRawHitMaker). Dmitry and Jeff are added to the reviewer list.

ggfdsa10 commented:
I think these PR test errors are caused by StRHICfFunction class not being recognized in StMuDST.
should I separate the function like a checkGSOBarSize() from the BFC level?
If I should do that, I will revise the StRHICfUtil directory is used only for the analysis level (MuDst -> picoDst)

plexoos commented:
Seunghwan, for test purposes, can you provide commands (and input files) to convert raw and/or simulated data to StEven and muDst formats? Thanks

ggfdsa10 commented:
Hello Dmitry, 
I'm attached runBfc.C file link.
https://github.com/ggfdsa10/STAR-RHICf/blob/main/runBfc.C

"rhicfdat" BFC option is RHICf raw data saving option for convert from .daq to MuDst.
and that code includes one .daq file path. "st_rhicf_18178002_raw_1000004.daq"

Thank you.

Commit:
remove StRHICfFunction

ggfdsa10 commented:
Dear All,
I think all of the PR test actions are checkers about the process of BFC -> MuDst level.

As I said before, that error seems caused by StRHICfFunction class in StMuRHICfUtil class.
also, all of that actions do not include "RHICfUtil" lib that is "rhicfutil" BFC option. and that option is used by only RHICf process (that library is generated from the StRHICfUtil directory)

due to people who are not using RHICf do not need to call the RHICfUtil library, separating between StMuDST and StRHICfUtil is necessary 

For this reason, I have removed the StRHICfFunction class header and functions at StMuRHICfUtil class.
But, In the "StRHICfRawHitMaker", exist the StRHICfFunction yet.

I'm not sure if I can upload the Pull Request again myself. 
but I would like to request the revised StMuRHICfUtil class again. 
I'm so sorry if you already reviewed this part.

Thank you

klendathu2k commented on StRoot/StBFChain/doc/index.html:
```diff
@@ -711,17 +711,19 @@ <H1 ALIGN="center">BFChain Options</H1>
 <tr><td><A NAME="STEVENT0">StEvent</A></td><td>&nbsp;</td><td>&nbsp;</td><td><A HREF="#GLOBT0">globT</A> <A HREF="#SCL0">SCL</A> <A HREF="#TRGDEF0">TRGDef</A> <A HREF="#STBICHSEL0">StBichsel</A> <A HREF="#EMCUTIL0">EmcUtil</A> <A HREF="#TBUTIL0">TbUtil</A> <A HREF="#DETDB0">detDb</A> </td><td>&nbsp;</td><td>StEvent</td><td>Load StEvent</td></tr>
```
StBFChain/doc contains files (such as this one) which are automatically generated at compile time, and should not be added to this pull request.

klendathu2k commented on StRoot/StEvent/StRHICfCollection.h:
```diff
@@ -1,71 +1,64 @@
+		unsigned int mRunTime[kRHICfNorder];
+		unsigned int mRunTRGM;
 
 	ClassDef(StRHICfCollection,1)
```
Should update the classdef to `ClassDef(StRHICfCollection, 2)`

klendathu2k commented on StRoot/StEvent/StRHICfCollection.h:
```diff
@@ -1,71 +1,64 @@
+		StRHICfHit* mRHICfHitColl; 
+		std::vector<StRHICfPoint*> mRHICfPointColl; 
+
+		unsigned int mRunNumber;
```
StEventInfo::mRunId and StEventInfo::mId provide _the_ run and event number for each entry in StEvent.  I am not sure why these are necessary.

klendathu2k commented on StRoot/StEvent/StRHICfRawHit.h:
```diff
@@ -0,0 +1,44 @@
+		Int_t mPlateADCDelay[kRHICfNtower][kRHICfNplate][kRHICfNrange]; // Delayed ADC of GSO plate (for pedestal)
+		Int_t mGSOSmallADC[kRHICfNlayer][kRHICfNxy][kRHICfNbarSmall];   // ADC of GSO bar of small tower 
+		Int_t mGSOLargeADC[kRHICfNlayer][kRHICfNxy][kRHICfNbarLarge];   // ADC of GSO bar of large tower
+		unsigned int mTDC[kRHICfNtdc];   // TDC 
```
You're mixing root types (Int_t) and standard c++ types (unsigned int).  For IO classes (such as this one) we prefer to use the ROOT types, per STAR coding guidelines.  `unsigned int` should be `UInt_t`

klendathu2k commented on StRoot/StEvent/StRHICfRawHit.h:
```diff
@@ -0,0 +1,44 @@
+		unsigned int getGPI1(Int_t idx);
+
+	private:
+		Int_t mPlateADC[kRHICfNtower][kRHICfNplate][kRHICfNrange];      // ADC of GSO plate
```
In terms of size, do you really need a full Int_t here?  Could you fit ADC values into a short?

ggfdsa10 replied:
I think UShort_t would be the best, Thank you

klendathu2k commented on StRoot/StRHICfUtil/StRHICfFunction.h:
```diff
@@ -0,0 +1,72 @@
+        void initChecker();
+
+		void setRunType(Int_t type){mRunType = type;}
+		Int_t getRunType(){return mRunType;}
```
This appears to be a utility class, rather than a persistent data class.  The preference is to use standard c++ types (int, float, etc...) rather than the ROOT tyles (Int_t, Float_t, etc...).

Also, because this class is a persistent class (i.e. saved to disk) you do not need the ClassDef statement below.

ggfdsa10 replied:
This class is a utility class, not a saving class.

**Also, because this class is a persistent class (i.e. saved to disk) you do not need the ClassDef statement below.**

as per your comment, if a class is "a utility class", there is no need for the ClassDef, it's right?

klendathu2k replied:
Correct.  If you do not write it to disk (e.g. it is not part of the MuDst or StEvent) then you do not need to declare the ClassDef.

ggfdsa10 replied:
Thank you for the answer.

klendathu2k reviewed (changes requested):
I don't see any major issues at this point.  I've noted a few things in the event data model which I would like to see addressed.  After they are attended to I may (or may not) have additional comments.

ggfdsa10 committed:
Delete index.html

remove from commit as per comments

ggfdsa10 committed:
Merge branch 'star-bnl:main' into main

Commit:
revised some class as per comments.

1. StRHICfCollection:
1) remove the parts of the run and event number.
2) change the data type using ROOT style (unsigned int -> UInt_t)
3) change the class def version 1 -> 2

2. StRHICfRawHit:
1) change the data type same as StRHICfCollection: 2).
2) change the data type for ADC (Int_t -> UShort_t)

3. StMuRHICfRawHIt:
1) change the data type same as StRHICfRawHit: 1) and 2).
2) remove the number same as StRHICfCollection: 1).

Commit:
remove the part of the saving run and event number

Commit:
tiny modification

dmarkh reviewed (approved):
StRHICfDbMaker looks good to me, no changes are required.

Commit:
revised StRHICfFunction

1. change the data type from ROOT style to C++ style
2. remove ClassDef and ClassImp

jml985 commented:
It appears that I was removed as a reviewer (which is fine with me).   I don't have any opinions or insight into all the failed CE tests.  

I did however take a look at the StRHICfDbMaker and StRHICfHitMaker additions.    I think the Db Maker probably meets the normal standards just fine.   The tables are filled once at the beginning of each run, and if there were more than one run would be memory leaks, but I think this is impossible for the offline chain.  The function for deleting tables is empty.   The only code is the setup in the initRun() and a bunch of helper functions translating the DB data into detector specific items.  I don't see any structural issues there.

In the stRHICfHitMaker codes, the setup logic in lines 56-79 is very complex and the actual logic is very simple.   The actual logic appears to be a simple linear assignment of variables each dependent on previous step:

if(!eventPtr)  return kStFatal;

mRHICfCollection = eventPtr->rhicfCollection();
if(!mRHICfCollection) mRHICfCollection=new StRHICfCollection;
if(!mRHICfCollection) return kStFatal;

mRHICfRawHitColl=mRHICfCollection->rawHitCollection();
if(!mRHICfRawHitColl) return kStFatal;

But the existing is logically unclear, and the setupRHICfCollection() function does nothing except make it less clear.

starsdong commented:
Hi Jeff, I guess you were probably accidentally removed. I added you back in.
For everyone, Dmitry A. and Jeff L. were asked and agreed to review the new two makers: StRHICfDbMaker and StRHICfRawHitMaker as our normal review procedure for new codes.

jml985 reviewed (changes requested):
See my previous comment.   The suggested change is to simplify the logic in StRICHfRawHitMaker lines 56-79.   All three variables must be initialized to proceed, and any problem initializing them causes a fatal error.  This is obscured by the nested if statements.

ggfdsa10 commented:
I'm so sorry for my mistake, I don't know why I removed a reviewer.
I guess there is something wrong when I clicked the re-request button

anyway, I don't know am I understanding your comments.
but, as far as I understood your comments, 
StRHICfRawHitMaker needs to simplify a code. such as your comments:

**if(!eventPtr) return kStFatal;

mRHICfCollection = eventPtr->rhicfCollection();
if(!mRHICfCollection) mRHICfCollection=new StRHICfCollection;
if(!mRHICfCollection) return kStFatal;

mRHICfRawHitColl=mRHICfCollection->rawHitCollection();
if(!mRHICfRawHitColl) return kStFatal;**

Also, that have to get to need the **eventPtr->setRHICfCollection(mRHICfCollection)**
because there is no RHICfCollection at all of the event starting points.
I've already checked when that code is present or not present.

and then, 
as per your other comment, StRHICfDbMaker is well done but should remove unnecessary functions like the deleteArrays().
If I was wrong, please let me know.

Thank you!

Commit:
remove the deleteArrays()

Commit:
revised Collection setop part

ggfdsa10 commented:
oh... I'm very sorry for Jason. when I clicked the re-requested button, removed a reviewer again. :(
I will NOT click this button.

jml985 reviewed (approved):
This update is much more clear, thank you.

plexoos committed:
ci: Add test for RHICf

starsdong commented:
Hi All,
The two new makers are reviewed and approved by reviewers. We need to follow up on other changes in this PR too.
@klendathu2k, on the StEvent containers, do you have any further comments/suggestions?
@genevb, any request on the BFChain options?
@jdbrice, could you please take a look at the MuDst changes?
THanks and Best Regards

genevb commented on StRoot/StRHICfUtil/StRHICfFunction.h:
```diff
@@ -0,0 +1,73 @@
+
+using namespace std;
+
+class StEnumerations;
```
Is this line needed if you have above:

    #include "StRoot/StEvent/StEnumerations.h"

?

ggfdsa10 replied:
Hello Gene, 

There are some functions that check whether there is correct input data or not. (this is just checking the size of the data)
So some parameters like kRHICfNtower, kRHICfNlayer, and so on are in this source file.

If someone uses the RHICf reconstruction tool without changing something, these checking functions will be meaningless.

If you think not needed these checking functions, then I will remove these functions and StEnumerations.hh include line.

Thank you

genevb replied:
Thanks for your reply, but I think I was not clear. I am saying that _if_ you have the `#include` statement, you _do_not_need_ the `class StEnumerations;` statement because the StEnumerations class is already known from the `#include` statement. You can delete the `class StEnumerations;` line.

genevb replied:
And there is more...

I do not see the use of any functionality defined in `StEnumerations.h` here in `StRHICfFunctions.h`. So one question to ask is, "Why have the `#include` for `StEnumerations.h` inside `StRHICfFunctions.h`?"

You should answer this question.

One possible answer is that every other code that includes `StRHICfFunctions.h` also needs functions defined in `StEnumerations.h`, so this is a shortcut to avoid having to include `StEnumerations.h` in those other codes.

Another possible answer is that it may be better to change the codes to include `StEnumerations.h` only in those other codes that really need it, and remove both the `class` and the `#include` for `StEnumerations` from `StRHICfFunctions.h`.

I hope that is clear, but please let me know if I need to explain the question better.

Thanks,
-Gene

ggfdsa10 replied:
class StEnumerations; this line is my mistake. sorry that is ok if removed.
I will remove it.

and I mean, some constant values for RHICf in the **StEnumerations.h** are used in the **StRHICfFunctions.cxx** source file.
In the **StRHICfFunctions.cxx**, you can see some values like a kRHICfNtower, kRHICfNlayer, etc.. 

the **StRHICfFunction.h** class is called in other classes such as **StRHICfHitMaker**, and so on.
so as you say, the **StEnumeration** class should be better to call only one time using the **StRHICfFunction** in the RHICf library
Likewise, I will remove including part for **StEnumeration.h** in other classes that already inherited **StRHICfFunction** 

Thank you for your comments.

ggfdsa10 replied:
I had committed the code. Thank you.

genevb commented on StRoot/StBFChain/BigFullChain.h:
```diff
@@ -1712,6 +1714,13 @@ Bfc_st BFC[] = { // standard chains
+  {"rhicfCali" ,"","", "StEvent,RhicfUtil,rhicfdb",
+   "StRHICfHitMaker","StRHICfHitMaker","RHICf calibration", kFALSE},
+  {"rhicfRec" ,"","", "StEvent,RhicfUtil,rhicfdb",
+   "StRHICfPointMaker","StRHICfPointMaker","RHICf reconstruction", kFALSE},
```
I don't see codes for StRHICfHitMaker and StRHICfPointMaker. Am I missing something? Are those coming later? If so, I think the chain options should be introduced when those are introduced.

ggfdsa10 replied:
Yes, StRHICfHitMaker and StRHICfPointMaker will be added later. (But these makers have already been made)
I will make new Pull-requests for calibration and reconstruction makers, including these BFC options later.

klendathu2k commented:
Hi Xin,

I see many of my comments resolved.  But there are some memory 
allocations that need
to be cleaned up.

Cheers,
Jason

On 2023-01-05 13:11, Xin Dong wrote:
> Hi All,
> The two new makers are reviewed and approved by reviewers. We need to
> follow up on other changes in this PR too.
> @klendathu2k [1], on the StEvent containers, do you have any further
> comments/suggestions?
> @genevb [2], any request on the BFChain options?
> @jdbrice [3], could you please take a look at the MuDst changes?
> THanks and Best Regards
> 
> --
> Reply to this email directly, view it on GitHub [4], or unsubscribe
> [5].
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
> 
> Links:
> ------
> [1] https://github.com/klendathu2k
> [2] https://github.com/genevb
> [3] https://github.com/jdbrice
> [4] 
> https://github.com/star-bnl/star-sw/pull/460#issuecomment-1372562548
> [5] 
> https://github.com/notifications/unsubscribe-auth/ANL4LVEAZX5NZDV3D44N3VDWQ4FGLANCNFSM6AAAAAASYHPUV4

plexoos commented:
> I'm attached runBfc.C file link. https://github.com/ggfdsa10/STAR-RHICf/blob/main/runBfc.C
> 
> "rhicfdat" BFC option is RHICf raw data saving option for convert from .daq to MuDst. and that code includes one .daq file path. "st_rhicf_18178002_raw_1000004.daq"

The daq file and the chain options have been added to CI tests. See #476 I would recommend to incorporate this along with modifications to StEvent and StRHICfRawHitMaker because that's what `rhicfdat` option depends on.

ggfdsa10 commented:
Hello Jason, 
Please tell me if there is anything I need to revise the code to arrange the parts of the allocation memory.
Thank you

ggfdsa10 commented:
Hello Dmitri,
I don't understand your comments.
Do you mean, I just need to add this macro https://github.com/ggfdsa10/STAR-RHICf/blob/main/runBfc.C in this pull request?

If I don't need to additional revise some BFC options, I will solve Gene's request as below. 

Gene: "I don't see codes for StRHICfHitMaker and StRHICfPointMaker. Am I missing something? Are those coming later? If so, I think the chain options should be introduced when those are introduced."

Please let me know if I'm wrong.
Thank you

plexoos commented:
> I don't understand your comments.
> Do you mean, I just need to add this macro https://github.com/ggfdsa10/STAR-RHICf/blob/main/runBfc.C in this pull request?

No, you don't need to add your macro. In order to enable that test, you should add the changes from #476 (for example, by cherry-picking the commit) onto this PR branch. If you don't know how to do this, let me know and I'll do it.

Your test, as it is now, will simply make calls to StRHICfRawHitMaker and hopefully fill StEvent with RHICf data. If, however, you have a different macro that somehow verifies the content of the produced output files, I'd be happy to help you to accommodate it as a CI test.

> If I don't need to additional revise some BFC options, I will solve Gene's request as below.
> 
> Gene: "I don't see codes for StRHICfHitMaker and StRHICfPointMaker. Am I missing something? Are those coming later? If so, I think the chain options should be introduced when those are introduced." 

Yes, I agree with Gene, the options calling the non-existent makers should be removed from this PR and introduced along with those new makers in a separate PR. I would add that introducing tests for the new options is highly desirable.

plexoos committed:
Merge branch 'main' into main

ggfdsa10 commented:
Dear Dmitri,

Now I'm clear about what are you saying. 
I don't know how to add a CI test to this request. 
If you help me with this part I appreciate you.

Also, I share some macro file that is verifying for output MuDst file with you.
https://github.com/ggfdsa10/STAR-RHICf/blob/main/openMuDst.C

If this macro is something wrong or does not function, please tell me 
Also, if there's anything I need to do, please let me know anytime.

Thank you!

Commit:
removed RHICf Cali and Reco option

Commit:
revise the include parts

plexoos commented:
Hi Seunghwan,

I pushed the change enabling the CI test to this PR branch.
Please note that your code does not compile due to the following errors:

```
#10 2398.2 g++ -m64 -fPIC -pipe -Wall -Woverloaded-virtual -std=c++0x -Wno-long-long -pthread -Wno-deprecated-declarations -Werror -O2 -g -falign-loops -falign-jumps -falign-functions -Dsl79_gcc485 -D__ROOT__ -DNEW_DAQ_READER -I. -IStRoot/StRHICfRawHitMaker -IStRoot/RTS/src -IStRoot/RTS/include -IStRoot/RTS/trg/include -I. -IStRoot -I.sl79_gcc485/include -I/opt/software/linux-scientific7-x86_64/gcc-4.8.5/root-5.34.38-l3v6vso6qgojm4l2ctwjojs6trbt4hpn/include -c .sl79_gcc485/OBJ/StRoot/StRHICfRawHitMaker/StRHICfRawHitMaker.cxx -o .sl79_gcc485/OBJ/StRoot/StRHICfRawHitMaker/StRHICfRawHitMaker.o
#10 2399.1 .sl79_gcc485/OBJ/StRoot/StRHICfRawHitMaker/StRHICfRawHitMaker.cxx: In member function 'virtual Int_t StRHICfRawHitMaker::Make()':
#10 2399.1 .sl79_gcc485/OBJ/StRoot/StRHICfRawHitMaker/StRHICfRawHitMaker.cxx:70:23: error: 'class StRHICfCollection' has no member named 'setRunNumber'
#10 2399.1    mRHICfCollection -> setRunNumber(gendata[0]);
#10 2399.1                        ^
#10 2399.1 .sl79_gcc485/OBJ/StRoot/StRHICfRawHitMaker/StRHICfRawHitMaker.cxx:71:23: error: 'class StRHICfCollection' has no member named 'setEventNumber'
#10 2399.1    mRHICfCollection -> setEventNumber(gendata[1]);
#10 2399.1                        ^
```

Commit:
tiny modifications

ggfdsa10 commented:
Ok that is revised Thank you

ggfdsa10 commented:
Dear All,

I was wondering if you could tell me when we can receive the code review approval and merge this request if you get a chance.
After this review, We would like to produce all of the raw data to muDst.
Please let me know if there is anything I need to do to review.

Thank you for your convenience,
Seunghwan

plexoos commented on StRoot/StEvent/StRHICfHit.h:
```diff
@@ -0,0 +1,86 @@
+		Float_t mMultiPeakHeight[kRHICfNtower][kRHICfNlayer][kRHICfNxy][kRHICfNorder];
+		Float_t mMultiPeakRaw[kRHICfNtower][kRHICfNlayer][kRHICfNxy][kRHICfNorder];
+		Float_t mMultiEnergySum[kRHICfNtower][kRHICfNlayer][kRHICfNxy][kRHICfNorder];
+		Float_t mMultiChiSquare[kRHICfNtower][kRHICfNlayer][kRHICfNxy];
```
That is an addition of 18 arrays some of which are 3 and 4 dimensional. Have you checked by how much the size of event.root files can increase on average?  These are static arrays so, if I think about this right all production chains will be affected, not just RHICf streams (if there is such thing) On the other hand, if they are empty the automatic ROOT compression should minimize the effect

ggfdsa10 replied:
Hello Dmitri,
Thank you for your reply, and I understand about this delayed PR Thank you.

We including the RHICf collaborator and I already had a discussion about these variable arrays.
Actually, In the reconstruction process of the RHICf library, 
If some events not satisfying a condition (for example, there is no worth for reconstruction or no have a tower hit),
then We don't store these arrays (except mPlateE, mGSOBarSmallE, mGSOBarLargeE, mL20, and mL90)
But our collaborators told me these arrays, all of the various arrays might be needed.

In spite of that, I know what do you concern about.
As I tell you above, these arrays only store a few events per 100 events. 
(But if there is a satisfying saving condition, all of the arrays are stored fully.)
So I think I have to change these array formats from static array to vector.

Please let me know if these changes are inappropriate.

Thank you,
Seunghwan

plexoos commented on StRoot/StRHICfDbMaker/StRHICfDbMaker.h:
```diff
@@ -0,0 +1,101 @@
+		TH2D* h2_TLphoLeakIn[15];
+
+	virtual const Char_t *GetCVS() const {static const Char_t cvs[]="Tag $Name:" __DATE__ " " __TIME__ ; return cvs;}
+	ClassDef(StRHICfDbMaker,3)   //StAF chain virtual base class for Makers
```
It is a maker. We don't version makers for IO

```suggestion
	ClassDef(StRHICfDbMaker, 0)
```

ggfdsa10 replied:
Thank you for pointing that out.

plexoos reviewed (commented):
Hi Seunghwan,

Sorry for the delay, but unfortunately it is something you can expect when you submit so many changes and include many new files in various packages. Usually making smaller changes more often is more efficient than dumping everything at once. It also helps to target specific packages/maintainers. For example, if you submitted the StRHICfDbMaker in a separate PR it could have been merged as I see that the maintainers already responded and approved. It is not late to do so, for instance, with the MuDst code.

Another general comment is that you should reformat the code to match the STAR coding style guidelines. In particular using tabs for indentation is not allowed. The easiest way to do this would be to run clang-format over the new files.

ggfdsa10 committed:
Update StRoot/StRHICfDbMaker/StRHICfDbMaker.h

Changes ClassDef version

Co-authored-by: Dmitri Smirnov <dmixsmi@gmail.com>

ggfdsa10 commented:
Hello Dmitri, 

Thank you for your comments and sorry for I didn't follow the STAR guidelines.
I think all of the files in this PR have to revise part of the indentation. 
Also, In my PR files, Is there any other STAR coding guideline that I'm missing?
I would appreciate it if you could let me know.

Seunghwan

Commit:
modifying indentation tap to space

Commit:
StRHICfDbMaker re-revised

Commit:
DbMaker commit version change

Commit:
DbMaker change the indentation...

Commit:
change parameter init

Commit:
add the function 'addPointCollction()'

Commit:
Modify the RHICf Hit data format:
  n-dimenssion arrays (e.g. singleHitPos[tower][layer][xy]) are modify to dynamic array (TArrayF class)

  StRHICfHit and StMuRHICfHit are added function that works save or not.
  therefore, if someone want to save these detailed datas, trun-on saveDetailInfo(true) in StRHICfPointMaker
  (defualt: not saving these data)

  StMuRHICfUtil:
   change the input parameter of some functions (fillRHICfRawHit,fillRHICfHit etc..)
   and fillRHICfHit, filleMuRHICfHit function were modified by StRHICfHit and StMuRHICfHit.

ggfdsa10 commented:
Dear All, 

I'm sorry for being late.
now I modified the indentation of all code and n-dimension data arrays in StRHICfHit, StMuRHICfHit according to comments.

These commits were designed more can be reduced memory.
also, when someone wants detailed n-dimension data in StRHICfHit, he can obtain that immediately.

Please review these codes and the remaining codes if you get a chance.

Seunghwan

plexoos commented on StRoot/StRHICfUtil/StRHICfFunction.cxx:
```diff
@@ -22,8 +22,11 @@ void StRHICfFunction::initChecker()
+  for(int i=0; i<5; i++){
+    if(i<3){mLeakageNum[i] = 0;}
+    mValueForRecoNum[i] = 0;
+  }
```
Why this change?
In C++ you probably want to zero C-arrays like this:

```suggestion
std::fill(mLeakageNum, mLeakageNum+3, 0);
std::fill(mValueForRecoNum, mValueForRecoNum+5, 0);
```

ggfdsa10 replied:
Thank you for the suggestion.
I thought FOR loop was more effective.
I commit again

plexoos reviewed (approved):
It'd be nice to verify the performance with tests but if there is no intention to add them it's fine by me

ggfdsa10 committed:
Update StRoot/StRHICfUtil/StRHICfFunction.cxx

change back commit FOR loop to std::fill

Co-authored-by: Dmitri Smirnov <dmixsmi@gmail.com>

plexoos commented:
If there are no further comments, is it okay to merge this PR?

plexoos committed:
Merge branch 'main' into main

ggfdsa10 commented:
Thank you for merging! 
However, there seems to be a problem with the CI test docker.
Please let me know if there is anything I can do to fix this problem.

Sincerely,
Seunghwan

plexoos commented:
Seunghwan, there was a problem with GitHub package repository access. They seem have fixed the problem and I restarted the jobs

Commit:
add a function of pointers deleting

ggfdsa10 commented:
Dear All, 

New change is commit for solve the last change requested one.
I'm so sorry for the inconvenience caused by committing after the commit request.

Seunghwan.

plexoos commented:
Seunghwan, just create a new PR with fixes. Thanks

ggfdsa10 commented:
Dear All, 

I created a new pull request for Dmitry's request. #511 
I'm very sorry for the inconvenience.

Seunghwan

plexoos commented:
Oh, so this one was not really merged... Ok, seems like another chance to review the code in #511
